bytecomp/bytelink.ml

```diff
@@ -212,39 +192,24 @@ let implementations_defined = ref ([] : (CU.Name.t * string) list)
 
 let check_consistency file_name cu =
   begin try
-<<<<<<< HEAD
+    let source = List.assoc (CU.name cu.cu_name) !implementations_defined in
+    raise (Error (Multiple_definition(cu.cu_name |> CU.full_path_as_string, file_name, source)));
+  with Not_found -> ()
+  end;
+  begin try
     Array.iter
       (fun import ->
         let name = Import_info.name import in
         let crco = Import_info.crc_with_unit import in
-||||||| merged common ancestors
-    List.iter
-      (fun (name, crco) ->
-=======
-    let source = List.assoc cu.cu_name !implementations_defined in
-    raise (Error (Multiple_definition(cu.cu_name, file_name, source)));
-  with Not_found -> ()
-  end;
-  begin try
-    List.iter
-      (fun (name, crco) ->
->>>>>>> ocaml/5.1
         interfaces := name :: !interfaces;
         match crco with
           None -> ()
-<<<<<<< HEAD
         | Some (full_name, crc) ->
             if CU.Name.equal name (CU.name cu.cu_name)
-            then Consistbl.set crc_interfaces name full_name crc file_name
-            else Consistbl.check crc_interfaces name full_name crc file_name)
-||||||| merged common ancestors
-        | Some crc ->
-            if name = cu.cu_name
-            then Consistbl.set crc_interfaces name crc file_name
-            else Consistbl.check crc_interfaces name crc file_name)
-=======
-        | Some crc -> Consistbl.check crc_interfaces name crc file_name)
->>>>>>> ocaml/5.1
+            (* CR mshinwell for lmaurer: please double-check, these two used
+               to be [Consistbl.set] and [Consistbl.check] respectively *)
+            then Consistbl.check crc_interfaces name full_name crc file_name
+            else Consistbl.check_noadd crc_interfaces name full_name crc file_name)
```

Comment:
I don't know any reason we can't just accept the upstream change here, namely just write
```ocaml
Consistbl.check crc_interfaces name full_name crc file_name
``` 
in place of the whole if/then/else. Notably, this is what #174 currently does: https://github.com/ocaml-flambda/ocaml-jst/blob/60d0234646097c947f5bed057a9f9d51a2eaa5c7/asmcomp/asmlink.ml#L60

Reply:
Fixed